tests/state/Areas/AreaTreeLevels/selectors-test.js

```diff
@@ -0,0 +1,241 @@
+							byKey: null,
+						},
+					},
+				},
+				expectedResult: [],
+			},
+			{
+				name: 'empty',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {},
+						},
+					},
+				},
+				expectedResult: [],
+			},
+			{
+				name: 'some',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {
+								k1: {n: 1},
+								k2: {n: 2},
+								k3: {n: 3, removed: true},
+							},
+						},
+					},
+				},
+				expectedResult: [{n: 1}, {n: 2}],
+			},
+		];
+
+		tests.forEach((test) => {
+			it(test.name, function () {
+				assert.deepStrictEqual(
+					selectors.getAll(test.state),
+					test.expectedResult
+				);
+			});
+		});
+	});
+
+	describe('getAllAsObject', function () {
+		const tests = [
+			{
+				name: 'null',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {},
+						},
+					},
+				},
+				expectedResult: {},
+			},
+			{
+				name: 'empty',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {},
+						},
+					},
+				},
+				expectedResult: {},
+			},
+			{
+				name: 'some',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {
+								k1: {n: 1},
+								k2: {n: 2},
+								k3: {n: 3, removed: true},
+							},
+						},
+					},
+				},
+				expectedResult: {
+					k1: {n: 1},
+					k2: {n: 2},
+				},
+			},
+		];
+
+		tests.forEach((test) => {
+			it(test.name, function () {
+				assert.deepStrictEqual(
+					selectors.getAllAsObject(test.state),
+					test.expectedResult
+				);
+			});
+		});
+	});
+
+	describe('getAllForActiveScope', function () {
+		const tests = [
+			{
+				name: 'empty models',
+				state: {
+					areas: {
+						areaTreeLevels: {},
+					},
+					scopes: {
+						activeKey: 'scopeK',
+					},
+				},
+				expectedResult: null,
+			},
+			{
+				name: 'empty indexes',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {k1: {n: 1}, k2: {n: 2}},
+							indexes: [],
+						},
+					},
+					scopes: {
+						activeKey: 'scopeK',
+					},
+				},
+				expectedResult: null,
+			},
+			{
+				name: 'no active skope key',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {k1: {n: 1}, k2: {n: 2}},
+							indexes: [
+								{filter: 'fil', order: 'desc'},
+								{
+									filter: {scopeKey: 'scopeK'},
+									order: 'asc',
+									count: 4,
+									index: [null, 'k1', 'k2', 'k3'],
+								},
+							],
+						},
+					},
+					scopes: {},
+				},
+				expectedResult: null,
+			},
+			{
+				name: 'some',
+				state: {
+					areas: {
+						areaTreeLevels: {
+							byKey: {k1: {n: 1}, k2: {n: 2}},
+							indexes: [
+								{filter: 'fil', order: 'desc'},
+								{
+									filter: {scopeKey: 'scopeK'},
+									order: 'asc',
+									count: 4,
+									index: [null, 'k1', 'k2', 'k3'],
+								},
+							],
+						},
+					},
+					scopes: {
+						activeKey: 'scopeK',
+					},
+				},
+				expectedResult: [{n: 1}, {n: 2}, {key: 'k3'}, null],
```

Comment:
@tmdmc I need your help here. In my opinion, expected result should be: [null, {n: 1}, {n: 2}, {key: 'k3'}], what do you think?